packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -94,6 +94,9 @@ class ServiceConnectionManager {
 
   ConnectedApp? connectedApp;
 
+  AppState get appState => _appState!;
+  AppState? _appState;
```

Comment:
instead of making this it's own member on serviceManager, can we move it into ConnectedApp? That way it is clear that this state is for the connected app and not the DevTools app. This would read nicer and tightly couple the AppState instance with the ConnectedApp instance.

Reply:
Yes, I started with it. The problem is that debugger sometimes uses the values of AppState when the ConnectedApp is null.

Most likely it is bug of our features and/or test mocking, but it will be significant effort to fix.
Added TODO.